### PR TITLE
feat(sdk): deprecate .add_node_selector_constraint in favor of .set_accelerator_type

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Deprecates pipeline task `.add_node_selector_constraint` in favor of `.set_accelerator_type`, 
since accelerators can be set on multiple platforms.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about 
the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/components/pipeline_task.py [hunk 1]
```diff
@@ -401,8 +401,21 @@ def add_node_selector_constraint(self, accelerator: str) -> 'PipelineTask':
...
+        warnings.warn(
+            f'{self.add_node_selector_constraint.__name__!r} is deprecated. Please use {self.set_accelerator_type.__name__!r} instead.',
+            category=DeprecationWarning)
+        return self.set_accelerator_type(accelerator)
+
+    def set_accelerator_type(self, accelerator: str) -> 'PipelineTask':
+        """Sets accelerator type to use when executing this task.
+
+        Args:
+            value: The name of the accelerator, such as ``'NVIDIA_TESLA_K80'`` and ``'TPU_V3'``.
```
Comment (gkcalat) [inline sdk/python/kfp/components/pipeline_task.py:418]: For KFP users, they should specify the vendor of the accelerator. Valid values generally vary across platfoms, but the popular ones are: `nvidia.com/gpu`, `amd.com/gpu`, `cloud-tpus.google.com/v2`, `cloud-tpus.google.com/v3`.

I propose to mention at least `nvidia.com/gpu` and `cloud-tpus.google.com/v3` as OSS counterparts to `NVIDIA_TESLA_K80` and `TPU_V3`.

**Review by gkcalat — COMMENTED** [review]
Thank you @connor-mccarthy!
I propose to update the docstring for KFP users.

File: sdk/python/kfp/components/pipeline_task.py (hunk 1 above)
Comment (connor-mccarthy) [reply sdk/python/kfp/components/pipeline_task.py]: Good thinking! Done.

**Review by gkcalat — APPROVED** [review]

**Comment by gkcalat** [comment]
/lgtm
/approve

Thanks, @connor-mccarthy!

**Pull request merged**